### DISCUSSION TITLE
how set 'q-page-container' children the remaining height

### DISCUSSION THREAD

**Discussion by jianyunli** [quasarframework/quasar#11785] [General - Components / Directives / etc]
I used a standard Header, Main, Footer layout, however the leaflet Map does not take the remaining height (full height - header - footer), the user have to scroll to bottom to see my map.  So I have to manfully set a height `style="height: 85vh;"` , this is ugly. 

question 1, how in the Map component, how do I know the exact header/footer height?
question 2, is there better way to do it?

many thanks

`<template>
  <q-layout view="hHh lpR fFf">

    <q-header elevated class="bg-primary text-white">
      <q-toolbar>
        <q-toolbar-title>
          <q-avatar>
            <img src="https://cdn.quasar.dev/logo-v2/svg/logo-mono-white.svg">
          </q-avatar>
          Title
        </q-toolbar-title>
      </q-toolbar>
    </q-header>

    <q-page-container>
      <Map />
    </q-page-container>

    <q-footer elevated class="bg-grey-8 text-white">
      <q-toolbar>
        <q-toolbar-title>
          <q-avatar>
            <img src="https://cdn.quasar.dev/logo-v2/svg/logo-mono-white.svg">
          </q-avatar>
          <div>Title</div>
        </q-toolbar-title>
      </q-toolbar>
    </q-footer>

  </q-layout>
</template>`

**Comment by hawkeye64** [comment]
If you don't want it to scroll (although this may be problematic for mobile), and the header and footer are not using `reveal` property, then I'd do something like this where header and footer are 50px each):

```css
style="height: calc(100vh - 100px)"
```

**Comment by jianyunli** [comment]
Thank you, Hawkeye64! This is sort of what I did, I manfully set the leaflet div height to `height: 87vh;` to make the <Map />fit. Is there other smarter way to know the exact header/footer height?

**Reply by hawkeye64** [reply]
That I'm not sure. You could probably access them via a ref.

**Reply by metalsadman** [reply]
in lieu of what @hawkeye64 said above, you can try to leverage one of these quasar provided helpers/plugins/components https://quasar.dev/quasar-utils/dom-utils#introduction, https://quasar.dev/options/screen-plugin, https://quasar.dev/vue-components/resize-observer
